File: core/src/main/java/ua/com/zak/budgetswing/core/di/ApplicationComponent.java
package ua.com.zak.budgetswing.core.di;

import javax.inject.Singleton;

import dagger.Component;
import ua.com.zak.budgetswing.core.mvp.presenter.AboutPresenter;
import ua.com.zak.budgetswing.core.mvp.presenter.AccountsPresenter;
import ua.com.zak.budgetswing.core.mvp.presenter.AddCategoryPresenter;
import ua.com.zak.budgetswing.core.mvp.presenter.CategoriesPresenter;
import ua.com.zak.budgetswing.core.mvp.presenter.HomePresenter;
import ua.com.zak.budgetswing.core.mvp.presenter.MainPresenter;
import ua.com.zak.budgetswing.core.mvp.presenter.AddTransactionPresenter;
import ua.com.zak.budgetswing.core.mvp.presenter.TransactionsPresenter;

/**
 * @author zak <zak@swingpulse.com>
 */
@Singleton
@Component(
    modules = {
            DaoModule.class,
            NavigationModule.class
    }
)
public interface ApplicationComponent {
    void inject(AccountsPresenter presenter);

    void inject(AddTransactionPresenter presenter);

    void inject(MainPresenter mainPresenter);

    void inject(HomePresenter homePresenter);

    void inject(CategoriesPresenter categoriesPresenter);

    void inject(TransactionsPresenter transactionsPresenter);

    void inject(AboutPresenter aboutPresenter);

    void inject(AddCategoryPresenter addCategoryPresenter);
}

File: app/src/main/java/com/robotnec/budget/refactored/di/ApplicationComponent.kt
package com.robotnec.budget.refactored.di

import android.app.Application
import com.robotnec.budget.refactored.BudgetApplication
import com.robotnec.budget.core.mvp.presenter.*
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        AndroidSupportInjectionModule::class,
        AndroidComponentsInjectionModule::class
    ]
)
interface ApplicationComponent : AndroidInjector<BudgetApplication> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): ApplicationComponent
    }

    fun inject(presenter: AccountsPresenter)

    fun inject(presenter: AddTransactionPresenter)

    fun inject(presenter: MainPresenter)

    fun inject(presenter: CategoriesPresenter)

    fun inject(presenter: TransactionsPresenter)

    fun inject(presenter: AboutPresenter)

    fun inject(presenter: AddCategoryPresenter)

    fun inject(presenter: AddAccountPresenter)

    fun inject(presenter: CategoryOverviewPresenter)

    fun inject(presenter: CalculatorPresenter)
}